### DISCUSSION TITLE
Authentication

### DISCUSSION THREAD

**Discussion by ekanshuj** [toeverything/AFFiNE#6030] [General]
When are you planning to implement 2FA in Affine?

**Comment by forehalo** [comment]
We are currently reimplementing a brand new authentication system which will bring us the ability to use AFFiNE in offline mode.

After main functions successfully landed, and all tests pass, we will add the additional content to the auth, for instance 2FA and passkey.

They all planned to be landed in 0.14 if everything works fine.

**Reply by MFYDev** [reply]
Hi, just selfhosted 0.15.4, and seems like 2fa is still not added yet, or did I miss anything? Really need this function, thanks!

**Comment by hilsonp** [comment]
I hope self-hosters will be able to protect AFFiNE with oidc like Authelia.
Thank you !

**Comment by gitwittidbit** [comment]
I, too, would be interested in mfa.
